pkg/cloud/aws/provider.go

```diff
@@ -1036,6 +1038,19 @@ func (aws *AWS) populatePricing(resp *http.Response, inputkeys map[string]bool)
 					skusToKeys[product.Sku] = key
 					aws.ValidPricingKeys[key] = true
 					aws.ValidPricingKeys[spotKey] = true
+				} else if strings.Contains(product.Attributes.UsageType, "LoadBalancerUsage") && product.Attributes.Operation == "LoadBalancing:Network" {
```

Comment:
Thoughts on filtering for `product.Attributes.Operation == "LoadBalancing"` instead? According to AWS pricing this would be a standard Elastic Load Balancer.

```json
    "T3YPSF6NJ69JWWNM" : {
      "sku" : "T3YPSF6NJ69JWWNM",
      "productFamily" : "Load Balancer",
      "attributes" : {
        "servicecode" : "AmazonEC2",
        "location" : "US East (N. Virginia)",
        "locationType" : "AWS Region",
        "group" : "ELB:Balancer",
        "groupDescription" : "Standard Elastic Load Balancer",
        "usagetype" : "LoadBalancerUsage",
        "operation" : "LoadBalancing",
        "regionCode" : "us-east-1",
        "servicename" : "Amazon Elastic Compute Cloud"
      }
    },
```

Reply:
The current implementation of the costmodel only takes kubernetes Services of type Loadbalancer into account for calculation costs. Those are all Network Load Balancers nowadays. When using AWS's loadbalancer controller you can have it manage your kubernetes Ingresses as well and for that it will create an Application Load Balancers.

There used to be a product called Classic Load Balancer that has been [deprecated](https://aws.amazon.com/blogs/aws/ec2-classic-is-retiring-heres-how-to-prepare/) but I don't know how "Standard Load Balancer" falls within AWS's proposition to be honest.

Reply:
@skoef Do you happen to have any screenshots showing that LoadBalancers are always of type "Network Load Balancer" as opposed to "Application Load Balancer"? It would be good to attach that screenshot to the PR's description.

Also, confirming what you said. CostModel is setting the `kubecost_load_balancer_cost` metric here ([code ref](https://github.com/opencost/opencost/blob/cec0575fcd4469309b5d69346f623fc8067aab7a/pkg/costmodel/metrics.go#L574-L591)). When calculating the cost for this metric it primarily checks Services of Type "LoadBalancer" ([code ref](https://github.com/opencost/opencost/blob/cec0575fcd4469309b5d69346f623fc8067aab7a/pkg/costmodel/costmodel.go#L1341-L1379)).

Reply:
@thomasvn I can't provide a screenshot, but did find in the code for AWS loadbalancer controller that it's hardcoded to always use Loadbalancers of type Network ([code ref](https://github.com/kubernetes-sigs/aws-load-balancer-controller/blob/main/pkg/service/model_build_load_balancer.go#L87)) for Service based loadbalancers. So for me it made sense to filter specifically on prices for loadbalancers of type Network. Again, this controller itself is able to manage Application Loadbalancers as well, but those are managed through Ingress and opencost doesn't do any accounting on those (yet).

Looking further through the price specifications, there are 3 flavors of loadbalancers: Network, Application and "Standard", but AWS doesn't publicly offer a "Standard" product, so I assumed that that was about the Classic loadbalancers.

Reply:
@skoef I just checked my environments and found that my Nginx Ingress Controller creates a Classic/Standard LoadBalancer.

Regardless, I think your current implementation is a good improvement and am ok defaulting to Network LoadBalancers. I think the cost difference is minimal. For example in us-east-1 the hourly rate of a Classic vs Network LoadBalancer is `.025/hr` vs `0.0225/hr` respectively.

It can be confirmed by looking at the following pricing sheet https://pricing.us-east-1.amazonaws.com/offers/v1.0/aws/AmazonEC2/current/us-east-1/index.json. LoadBalancerNetwork = SKU PQ44QRGZYQ87XPN2. LoadBalancer = SKU T3YPSF6NJ69JWWNM.